File: database/database-annotation-processor/src/test/java/io/koraframework/database/common/annotation/processor/app/TestKoraApp.java
package io.koraframework.database.common.annotation.processor.app;

import org.mockito.Mockito;
import io.koraframework.common.KoraApp;
import io.koraframework.common.annotation.Root;
import io.koraframework.database.common.annotation.Query;
import io.koraframework.database.common.annotation.Repository;
import io.koraframework.database.jdbc.JdbcConnectionFactory;
import io.koraframework.database.jdbc.JdbcRepository;

@KoraApp
public interface TestKoraApp {
    @Repository
    interface TestRepository extends JdbcRepository {
        @Query("INSERT INTO table(value) VALUES (:value)")
        void abstractMethod(String value);
    }

    default JdbcConnectionFactory jdbcQueryExecutorAccessor() {
        return Mockito.mock(JdbcConnectionFactory.class);
    }

    @Root
    default Object mockLifecycle(TestRepository testRepository) {
        return new Object();
    }

}

File: database/database-symbol-processor/src/test/kotlin/io/koraframework/database/symbol/processor/app/TestKoraApp.kt
package io.koraframework.database.symbol.processor.app

import org.mockito.Mockito
import io.koraframework.common.annotation.KoraApp
import io.koraframework.common.annotation.Root
import io.koraframework.database.common.annotation.Query
import io.koraframework.database.common.annotation.Repository
import io.koraframework.database.jdbc.JdbcExecutor
import io.koraframework.database.jdbc.JdbcRepository

@KoraApp
interface TestKoraApp {
    @Repository
    interface TestRepository : JdbcRepository {
        @Query("INSERT INTO table(value) VALUES (:value)")
        fun abstractMethod(value: String?)
    }

    fun jdbcQueryExecutorAccessor(): JdbcExecutor {
        return Mockito.mock<JdbcExecutor>(JdbcExecutor::class.java)
    }

    @Root
    fun mockLifecycle(testRepository: TestRepository) = Any()
}
